sdk/src/transaction_context.rs

```diff
@@ -667,19 +848,111 @@ impl<'a> BorrowedAccount<'a> {
             )
             .unwrap_or_default()
     }
+
+    /// Returns true if the owner of this account is the current `InstructionContext`s last program (instruction wide)
+    pub fn is_owned_by_current_program(&self) -> bool {
+        self.instruction_context
+            .get_last_program_key(self.transaction_context)
+            .map(|key| key == self.get_owner())
+            .unwrap_or_default()
+    }
+
+    /// Returns an error if `get_data_mut` would fail
+    pub fn can_data_be_changed(&self) -> Result<(), InstructionError> {
+        if !self
+            .transaction_context
+            .is_early_verification_of_account_modifications_enabled()
+        {
+            return Ok(());
+        }
+        // Only non-executable accounts data can be changed
+        if self.is_executable() {
+            return Err(InstructionError::ExecutableDataModified);
+        }
+        // and only if the account is writable
+        if !self.is_writable() {
+            return Err(InstructionError::ReadonlyDataModified);
+        }
+        // and only if we are the owner
+        if !self.is_owned_by_current_program() {
+            return Err(InstructionError::ExternalAccountDataModified);
+        }
+        Ok(())
+    }
+
+    /// Returns an error if `can_data_be_changed`, `set_data` or `set_data_length` would fail
```

Comment:
nit: I think this comment and the comment for `can_data_be_changed` needs to be updated